api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -496,12 +496,42 @@ public void setDiagnoses(Set<Diagnosis> diagnoses) {
 	 * @since 2.4.0, 2.3.1
 	 */
 	public Set<Condition> getConditions() {
-		if (conditions == null) {
-			conditions = new LinkedHashSet<>();
+
+		return getAllConditions();
+	}
+
+	/**
+	 * Returns all conditions where conditions.encounterId = Encounter.encounterId In practice, this method should
+	 * not be used very often...
+	 *
+	 * @param includeVoided specifies whether or not to include voided Conditions
+	 * @return Returns the all Conditions.
+	 * <strong>Should</strong> not return null with null conditions set
+	 * <strong>Should</strong> get conditions
+	 */
+	public Set<Condition> getAllConditions(boolean includeVoided) {
```

Comment:
Just name this `getCondition(Boolean)`, there is no need for a different name.

I know you borrowed this from the logic for the obs getters, but obs is a different situation, the distinction between "obs" and "all obs" is "top level" vs "including nested in groups" obs.